app/services/irs_impact_analysis_csv_service.rb

```diff
@@ -0,0 +1,103 @@
+require 'csv'
+
+class IrsImpactAnalysisCsvService
+  CSV_FIELDS = %w{
+    legal_name
+    last_four_ssn
+    date_of_birth
+    zip_code
+    reached_gathering_docs
+    uploaded_documents
+    completed_intake
+    filed
+    service_type
+  }.freeze
+
+  CSV_HEADERS = CSV_FIELDS.map { |field| field.gsub(/\W/, "") }.freeze
+
+  # only grab intakes where the user consented.
+  def intakes
+    Intake.where.not(primary_consented_to_service_at: nil).includes(:ticket_statuses, :documents)
+  end
+
+  def generate_csv
+    CSV.generate(headers: csv_headers, write_headers: true) do |csv|
+      intakes.find_each(batch_size: 100) do |intake|
+        csv << csv_row(decorated_intake(intake))
+      end
+    end
+  end
+
+  def send_csv
+    # TODO: Whats the best way to download this document?
+  end
```

Comment:
Now that we have Mailgun integrated, it might be easiest to send this as an email attachment.